scene/animation/animation_tree.h

```diff
@@ -380,6 +390,37 @@ class AnimationTree : public Node {
 	Variant post_process_key_value(const Ref<Animation> &p_anim, int p_track, Variant p_value, const Object *p_object, int p_object_idx = -1);
 	virtual Variant _post_process_key_value(const Ref<Animation> &p_anim, int p_track, Variant p_value, const Object *p_object, int p_object_idx = -1);
 
+	struct Record {
+		List<AnimationNode::AnimationState> state_animation_states;
```

Comment:
Opinion of using the RingBuffer<AnimationNode::AnimationState> for this purpose?

Reply:
This type is same to `AnimationNode::State::animation_states` and use to store `state.animation_States`.
It maybe better that using `Vector<AnimationNode::AnimationState>` for this case.